allennlp/modules/attention/linear_attention.py

```diff
@@ -0,0 +1,92 @@
+import math
+
+import torch
+from torch.nn import Parameter
+from overrides import overrides
+from allennlp.modules.attention.legacy_attention import Attention
+from allennlp.nn import util
+from allennlp.nn.activations import Activation
+from allennlp.common.params import Params
+
+
+@Attention.register("linear")
+class LinearAttention(Attention):
+    """
+    This similarity function performs a dot product between a vector of weights and some
```

Comment:
This docstring needs to be updated, as this isn't a similarity function anymore.